src/test/groovy/graphql/execution/ConditionalNodesTest.groovy

```diff
@@ -183,6 +191,27 @@ class ConditionalNodesTest extends Specification {
 
         then:
         er["data"] == ["in": "in", "out": "out"]
+
+        // TODO: this test demonstrates that the GraphQLContext is missing for Pet
+        // and therefore, fields are being incorrectly included
+        when:
+        ei = ExecutionInput.newExecutionInput()
+                .graphQLContext(contextMap)
+                .query("""
+            query q {
+                in
+                pet {
+                  name
+                  favouriteSnack @featureFlag(flagName : "OFF")
+                } 
+            }
+        """
+                ).build()
+        er = graphQL.execute(ei)
+
+        then:
+        // TODO: favouriteSnack should not appear in the data
+        er["data"] == ["in": "in", "pet": ["name": "name"]]
```

Comment:
ConditionalNodesTest > integration test showing conditional nodes can be custom included FAILED
Condition not satisfied:
er["data"] == ["in": "in", "pet": ["name": "name"]]
     | |        |
     | |        false
     | [in:in, pet:[name:name, favouriteSnack:favouriteSnack]]
     ExecutionResultImpl{errors=[], data={in=in, pet={name=name, favouriteSnack=favouriteSnack}}, dataPresent=true, extensions=null}
         at graphql.execution.ConditionalNodesTest.integration test showing conditional nodes can be custom included(ConditionalNodesTest.groovy:214)